src/main/java/com/limechain/trie/structure/nibble/Nibbles.java

```diff
@@ -67,6 +68,15 @@ public static Nibbles of(int... nibbles) {
         return new Nibbles(Arrays.stream(nibbles).mapToObj(Nibble::fromInt));
     }
 
+    public static Nibbles fromLEBytes(byte[] bytes) {
+        Nibbles nibbles = new Nibbles(new byte[]{});
+        for (int i = 0; i < bytes.length; i++) {
+            nibbles.add(Nibble.fromInt((bytes[i] >> 4 & 0xf)));
+            nibbles.add(Nibble.fromInt((bytes[i] & 0xf)));
+        }
+        return nibbles;
+    }
```

Comment:
That's what `BytesToNibbles` is for. Ideally, I'd suggest we keep the Nibbles class itself behaving like a collection, thus extract generational logic outside... or at least that was my intention. You could replace any invocations of `Nibbles.fromLEBytes(arr)` with `new Nibbles(new BytesToNibbles(arr))` for now. 

There will be additional refactoring to the entire Nibbles class anyway, it's a bit messy as of right now :D